src/expressions/dataTypes/Value.ts

```diff
@@ -1,3 +1,42 @@
 export default class Value {
-	constructor(public type: string, readonly value: any) {}
+	constructor(public type: ValueType, readonly value: any) {}
 }
+
+export type ValueType =
```

Comment:
Very nice!
I think we can even split these up in atomic specific and all types, or does that cause more issues? e.g.
```js
export type AnyAtomicType =
	| 'xs:boolean'
	| 'xs:string' ...;
export type ValueType =
	AnyAtomicType
	| 'node()'
	| 'element()'
	| 'document()'
	| 'text'
	| 'comment()'
	| 'item()'
	| 'function(*)'
	| 'map(*)'
	| 'array(*)';
```

Reply:
You certainly can, but I'm not sure on the rules :) 
Would the idea be that the AtomicValue type could only take an AnyAtomicType? And Value would still take ValueType?

Reply:
Yes, that is indeed my idea. That way we prevent non atomic values ending up in an `AtomicValue`.
It doesn't have to be part of this PR btw, we can of course do further improvements like those sometime later.

Reply:
Cool, well i had a quick go and it does cascade/get out of hand pretty quickly! So probably one for later :)

Reply:
Cool, something for later it is!